dpnp/linalg/dpnp_algo_linalg.pyx

```diff
@@ -194,18 +194,25 @@ cpdef tuple dpnp_eig(utils.dpnp_descriptor x1):
     cdef DPNPFuncType param1_type = dpnp_dtype_to_DPNPFuncType(x1.dtype)
     cdef DPNPFuncData kernel_data = get_dpnp_function_ptr(DPNP_FN_EIG_EXT, param1_type, param1_type)
 
-    result_type = dpnp_DPNPFuncType_to_dtype(< size_t > kernel_data.return_type)
-
     x1_obj = x1.get_array()
 
+    cdef custom_linalg_2in_1out_func_ptr_t func = NULL
+    cdef DPNPFuncType return_type = DPNP_FT_NONE
+    if dpnp.issubdtype(x1_obj.dtype, dpnp.integer) and not x1_obj.sycl_device.has_aspect_fp64:
+        return_type = kernel_data.return_type_no_fp64
+        func = < custom_linalg_2in_1out_func_ptr_t > kernel_data.ptr_no_fp64
+    else:
+        return_type = kernel_data.return_type
+        func = < custom_linalg_2in_1out_func_ptr_t > kernel_data.ptr
```

Comment:
Part of the code is common for all impacted functions and can be moved into a separate function call. something like:
```
cdef (DPNPFuncType, void *) get_ret_type_and_func(x1_obj, DPNPFuncData kernel_data):
    if dpnp.issubdtype(x1_obj.dtype, dpnp.integer) and not x1_obj.sycl_device.has_aspect_fp64:
        return_type = kernel_data.return_type_no_fp64
        func = kernel_data.ptr_no_fp64
    else:
        return_type = kernel_data.return_type
        func = kernel_data.ptr
    return return_type, func

...

cpdef tuple dpnp_eig(utils.dpnp_descriptor x1):
    ...
    cdef (DPNPFuncType, void *) ret_type_and_func = get_ret_type_and_func(x1_obj, kernel_data)
    cdef custom_linalg_2in_1out_func_ptr_t func = < custom_linalg_2in_1out_func_ptr_t > ret_type_and_func[1]
    cdef DPNPFuncType return_type = ret_type_and_func[0]
    ...
```